dfastmi/gui/dialog_view.py

```diff
@@ -142,235 +175,324 @@ def create_qt_application(self) -> None:
+        self._slength = QLabel(self._win)
+        self._slength.setToolTip(self._view_model.gui_text("length_tooltip"))
+        self._slength.setText(self._view_model.slength)
+        layout.addRow(self._view_model.gui_text("length"), self._slength)
+
+    def _create_ucritical_input(self, layout, double_validator):
+        self._ucrit = QLineEdit(self._win)
+        self._ucrit.setValidator(double_validator)
+        self._ucrit.setToolTip(self._view_model.gui_text("ucrit_tooltip"))
+        self._ucrit.editingFinished.connect(self._update_ucritical)
+        self._ucrit.setText(str(self._view_model.ucritical))
+        layout.addRow(self._view_model.gui_text("ucrit"), self._ucrit)
+
+    def _create_qthreshhold_input(self, layout, double_validator : PyQt5.QtGui.QDoubleValidator):
+        self._qthr = QLineEdit(self._win)
+        self._qthr.setValidator(double_validator)
+        self._qthr.editingFinished.connect(self._update_qthreshold)
+        print("Signal connected:", self._qthr.signalsBlocked())
+        self._qthr.setToolTip(self._view_model.gui_text("qthr_tooltip"))
+        qthrtxt = QLabel(self._view_model.gui_text("qthr"), self._win)
+        self._qthr.setText(str(self._view_model.qthreshold))
+        layout.addRow(qthrtxt, self._qthr)        
+
+    def _show_discharge_location(self, layout):
+        self._qloc = QLabel("", self._win)
+        self._qloc.setToolTip(self._view_model.gui_text("qloc"))
+        self._qloc.setText(self._view_model.current_branch.qlocation)
+        layout.addRow(self._view_model.gui_text("qloc"), self._qloc)
+
+    def _create_reach_input(self, layout):
+        # get the reach
+        self._reach = QComboBox(self._win)
+        self._reach.currentTextChanged.connect(self._view_model.updated_reach)
+        self._reach.setToolTip(self._view_model.gui_text("reach_tooltip"))
+        for r in self._view_model.current_branch.reaches:
+            self._reach.addItem(r.name)
+        self._reach.setCurrentText(self._view_model.current_reach.name)
+        layout.addRow(self._view_model.gui_text("reach"), self._reach)
+    
+    def _create_branch_input(self, layout:QFormLayout):
+        # get the branch
+        self._branch = QComboBox(self._win)
+        self._branch.currentTextChanged.connect(self._view_model.updated_branch)
+        self._branch.setToolTip(self._view_model.gui_text("branch_tooltip"))
+        for b in self._view_model.model.rivers.branches:
+            self._branch.addItem(b.name)
+        self._branch.setCurrentText(self._view_model.current_branch.name)
+        layout.addRow(self._view_model.gui_text("branch"), self._branch)
+
+    def _update_qthreshold(self):
+        if self._qthr.hasAcceptableInput():
+            self._view_model.qthreshold = float(self._qthr.text())
+            self._update_qvalues_table()
+            self._update_condition_files()
+        else: 
+            self._showMessage("Please input valid values for qthreshold")
```

Comment:
This feels like something that should definitely not be hardcoded in the gui.
Lets look into a followup issue.

Same below.

Reply:
may be we need to remove this popup?

Reply:
removed